source/Mlos.Python/mlos/Optimizers/RegressionModels/Prediction.py

```diff
@@ -89,6 +89,7 @@ def __init__(
             self.set_dataframe(dataframe)
 
     def set_dataframe(self, dataframe: pd.DataFrame):
+        dataframe = dataframe.astype(dtype=np.float64)

```

Comment:
Just assert that it's  a `float64` to begin with and fix all upstream code to comply.

Reply:
https://github.com/byte-sculptor/mlos_v2/issues/9